src/main/diagnostics/abstractFileActionProvider.ts

```diff
@@ -65,6 +67,17 @@ export abstract class AbstractFileActionProvider implements ExtensionComponent,
         return diagnostics;
     }
 
+    issueToDiagnostic(issue: CodeIssueTreeNode): vscode.Diagnostic {
+        let diagnostic: vscode.Diagnostic = new vscode.Diagnostic(
+            new vscode.Range(issue.regionWithIssue.start, issue.regionWithIssue.end),
+            issue.issueId,
+            vscode.DiagnosticSeverity.Warning
+        );
+        diagnostic.source = this.getSource();
+        diagnostic.code = 'Severity: ' + SeverityUtils.getString(issue.severity);
+        return diagnostic;
+    }
+
```

Comment:
this is redundant, you can use the createDiagnostics method.
I don't think we should create in the abstract class same method that is more specific for issues with region.
also the creation of the range is also redundant, issue.regionWithIssue is already a range.
I think you should remove this and use the general interface.
we can split the original method to createDiagnostic that only handles one location and return one diagnostic and than use it in the original createDiagnostics if you want to use the single one it in the map method that this method is used for.